src/widgets/FreeformInputWidget.tsx

```diff
@@ -0,0 +1,302 @@
+/* istanbul ignore file */
+/* See cypress/component/FreeformInputWidget.cy.tsx */
+import React, { useEffect, useMemo, useRef, useState } from 'react'
+import styled from 'styled-components'
+
+import {
+  Fieldset,
+  Legend,
+  ReservedSpace,
+  Widget,
+  WidgetHeader,
+  WidgetTitle,
+  Error
+} from './Widget'
+import { useBypassRequired } from '../utils'
+import { WidgetTooltip } from '..'
+import { useReadLocalStorage } from 'usehooks-ts'
+
+interface FreeformInputWidgetDetailsCommon {
+  comment?: string
+}
+
+interface FreeformInputWidgetDetailsFloat {
+  dtype: 'float'
+  default?: number
+}
+
+interface FreeformInputWidgetDetailsInteger {
+  dtype: 'int'
+  default?: number
+}
+
+interface FreeformInputWidgetDetailsString {
+  dtype: 'string'
+  default?: string
+}
+
+export type FreeformInputWidgetDetails = FreeformInputWidgetDetailsCommon &
+  (
+    | FreeformInputWidgetDetailsFloat
+    | FreeformInputWidgetDetailsInteger
+    | FreeformInputWidgetDetailsString
+  )
+
+export interface FreeformInputWidgetConfiguration {
+  type: 'FreeformInputWidget'
+  name: string
+  label: string
+  help?: string | null
+  required: boolean
+  details: FreeformInputWidgetDetails
+}
+
+interface FreeformInputWidgetProps {
+  configuration: FreeformInputWidgetConfiguration
+  /**
+   * Permitted selections for the widget.
+   */
+  constraints?: string[]
```

Comment:
I think we can get rid of this for now if constraints are not going to be already applied to this widget.